web/src/settings_profile_fields.ts

```diff
@@ -450,31 +510,31 @@ function open_edit_form_modal(e) {
             .on("input", ".choice-row input", add_choice_row);
         $profile_field_form
             .find(".edit_profile_field_choices_container")
-            .on("click", "button.delete-choice", (e) =>
-                delete_choice_row_for_edit(e, $profile_field_form, field),
-            );
+            .on("click", "button.delete-choice", function (this: HTMLElement) {
+                delete_choice_row.call(this);
+                disable_submit_btn_if_no_property_changed($profile_field_form, field);
+            });
```

Comment:
I've used `'call` to invoke the function with `this` value but I was wondering if it is a good approach, or if there is a better way to achieve this.